### PR TITLE
feat: support matching dot files

### PR DESCRIPTION
**Pull request by u3u**
I hope it can format and repair all matching files in my project, but I found that it did not format my configuration file (eg: .eslintrc.js, .postcssrc.js, etc.)

```json
"format": "prettier-eslint \"**/*.{js,json,vue,md}\" --write"
```

https://github.com/isaacs/node-glob#dots

### CODE REVIEW

**Comment by Oligrand** [comment]
@kentcdodds @zimme 
Any chance we could get this merged? Or at least give us the option to control the `dot` option of glob? It would be nice to also get files starting with a dot prettified :)

**Comment by kylemh** [comment]
@u3u feel free to resolve merge conflicts

@Oligrand feel free to open a new PR with the same fixes here if above doesn't happen fast enough for you!

**Comment by JounQin** [comment]
Dot files are excluded by ESLint and prettier by default AFAIK, maybe this new feature should be behind a flag?

**Comment by JounQin** [comment]
I think this should be implemented under a flag, can `false` by default same as ESLint and prettier.

File: src/format-files.js [hunk 1]
```diff
@@ -207,7 +207,7 @@ function getFilesFromGlob(
   applyPrettierIgnore,
   fileGlob
 ) {
-  const globOptions = { ignore: ignoreGlobs };
+  const globOptions = { dot: true, ignore: ignoreGlobs };
```
Comment (JounQin) [inline src/format-files.js:210]: TODO: add a flag instead.

**Review by JounQin — CHANGES_REQUESTED** [review]

**Comment by kylemh** [comment]
I think a `includeDotFiles` flag is good; however, you touched on a good point with the ignore files. If we WERE to respect those, how would we do it?

I can see at least 2 ways:
1. Use each ignore file for their respective runner. So, if ONLY a prettier ignore file exists, the eslint portion of our task will complete as expected and only the prettier portion would adhere to the ignore file.
2. We merge the ignore files and avoid those files for our entire runner.

**Comment by JounQin** [comment]
> Use each ignore file for their respective runner. 

I review the source codes we're using this approach now: https://github.com/prettier/prettier-eslint-cli/blob/master/src/format-files.js#L206-L236

So we'll only need to support a `includeDotFiles` flag.

**Comment by JounQin** [comment]
> Use each ignore file for their respective runner. 

I review the source codes we're using this approach now: https://github.com/prettier/prettier-eslint-cli/blob/master/src/format-files.js#L206-L236

So we'll only need to support a `includeDotFiles` flag.

**Comment by JounQin** [comment]
> Use each ignore file for their respective runner. 

I review the source codes we're using this approach now: https://github.com/prettier/prettier-eslint-cli/blob/master/src/format-files.js#L206-L236

So we'll only need to support a `includeDotFiles` flag.

**Review by JounQin — APPROVED** [review]

**Comment by JounQin** [comment]
TODO: document

**Review by kylemh — APPROVED** [review]

**Pull request merged**